lib/api/src/modules/stories.ts

```diff
@@ -72,6 +71,61 @@ export const init: ModuleFn = ({
   storyId: initialStoryId,
   viewMode: initialViewMode,
 }) => {
+  interface Meta {
+    ref?: ComposedRef;
+    source?: string;
+    sourceType?: 'local' | 'external';
+    refId?: string;
+    v?: number;
+    type: string;
+  }
+
+  function createEventHandlerWithRefSupport<T = { v?: number }>(
+    callback: (data: T, meta: Meta) => void
+  ) {
+    function handle(data: T) {
+      const { source, refId, type }: Meta = this;
+      const [sourceType, sourceLocation] = getSourceType(source, refId);
+      // @ts-ignore
+      const v = data.v || this.v;
+
+      if (v > 2) {
+        // eslint-disable-next-line no-console
+        console.warn(`Received ${this.type} event with version ${v}, we'll try and handle it`);
+      }
```

Comment:
This bit is specific to `SET_STORIES`

Reply:
Ow is it? I thought ALL events were going to be versioned?

Reply:
One day maybe.. but the version we support for a given event may change.. this is very much a event specific check.